driver/main.c

```diff
@@ -2808,39 +2811,45 @@ static int do_cpu_callback(unsigned long cpu, long sd_action) {
 #if(LINUX_VERSION_CODE >= KERNEL_VERSION(4, 10, 0))
 static int scap_cpu_online(unsigned int cpu) {
 	vpr_info("scap_cpu_online on cpu %d\n", cpu);
-	return do_cpu_callback(cpu, 1);
+	struct hotplug_st st;
+	st.sd_action = 1;
+	st.cpu = cpu;
+	return smp_call_on_cpu(0, do_cpu_callback, &st, true);
```

Comment:
Run the `do_cpu_callback` on the CPU 0 to make sure that we are able to send the `PPME_CPU_HOTPLUG_E` event to userspace.
This is what we do in the bpf probes, since we are guaranteed that CPU 0 is always present.